connectors/sources/sharepoint_online.py

```diff
@@ -1525,6 +1525,13 @@ def drive_item_operation(self, item):
             return OP_INDEX
 
     def download_function(self, drive_item, max_drive_item_age):
+        if "deleted" in drive_item:
+            self._logger.debug(
+                f"Not downloading the item id={drive_item['id']} because it has been deleted"
+            )
```

Comment:
Might be worth to add a note that `drive_item['id']` is not available for situation when file is deleted.